code/game/machinery/pipe/construction.dm

```diff
@@ -150,10 +151,16 @@
 			src.pipe_type = PIPE_OMNI_MIXER
 		else if(istype(make_from, /obj/machinery/atmospherics/omni/filter))
 			src.pipe_type = PIPE_OMNI_FILTER
+		else if(istype(make_from, /obj/machinery/atmospherics/binary/circulator))
+			src.pipe_type = PIPE_CIRCULATOR
 		
 		var/obj/machinery/atmospherics/trinary/triP = make_from
 		if(istype(triP) && triP.flipped)
 			src.flipped = 1
+		
+		var/obj/machinery/atmospherics/binary/circulator/circP = make_from
+		if(istype(circP) && circP.side == circP.CIRC_RIGHT)
+			src.flipped = 1

```

Comment:
You should have a look at the rest of the file :P. I'm keeping it here for consistency's sake.